### PR TITLE
Add contribution guidelines

### PR DESCRIPTION
**Pull request by BlakeWilliams**
Many PR's in the TIL repo lately have been more like micro-blog posts
rather than TIL's. This adds a "Contributing a TIL" section to help
define what a TIL is and what makes a good TIL.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,20 @@ you might also like
...
+## Contributing a TIL
+
+TIL's should be a focused and concise piece of knowledge that's interesting and
+worth sharing with others.
+
+Here's some guidelines on how to write a great TIL:
+
+* Avoid writing more than 60 lines of content.
+* Avoid writing more than 2 examples.
+* Avoid going into too much detail.
```
Comment (JoelQ) [inline README.md]: Thoughts on adding a guideline for not adding TILs that explicitly violate our guides?

Comment (BlakeWilliams) [inline README.md]: Yeah, that's a great idea.

Comment (BlakeWilliams) [inline README.md]: Added

**Comment by JoelQ** [comment]
Looks good to me. I'd love to get thoughts from @croaky @mjankowski @mike-burns

**Comment by mjankowski** [comment]
"A TIL is something which has too much detail to mention in daily standup, but is not long enough or interesting enough to be a blog post.  It's a bite-sized morsel of knowledge than you want to share with others."

Is that accurate?

**Comment by BlakeWilliams** [comment]
Yeah, that sounds accurate. They shouldn't go into too much detail, but should be somewhat general. If you go into too much detail you're writing a blog post.

File: README.md [hunk 2]
```diff
@@ -16,6 +16,23 @@ you might also like
 
 [Upcase]: https://upcase.com?utm_source=til
 
+## Contributing a TIL
+
+TIL's should be a focused and concise piece of knowledge that's interesting and
```
Comment (pirj) [inline README.md]: Do you think if it makes sense to add that TILs should be practically applicable rather than just worth sharing?

Comment (BlakeWilliams) [inline README.md]: I'm not sure. There's a lot of overlap of the two and I'd like to see where "interesting" would take us. Anyone else have thoughts on this?

**Comment by mike-burns** [comment]
Some questions I've had:

* Does the TIL have to be a thing that the person has learned today?
* What of TILs that we disagree with or do not advocate?

**Comment by BlakeWilliams** [comment]
I'd say ideally it'd be something they learned today, that would help keep it focused. I also think that we should close TIL's that we disagree with. It's `thoughtbot/til` and the tweets come from us, so it should only be content we agree with and feel comfortable sharing.

**Comment by JoelQ** [comment]
I don't think it absolutely has to be something learned today. It could be yesterday or during the past week. However, it shouldn't be for writing tips based on my experience in the past few years.

I do agree that we should restrict content to things we agree with since we are promoting it under your name. Do you think the guideline not to violate our guides is good enough?

**Comment by mike-burns** [comment]
Really it's "we decide whether to merge at our discretion", but that's not useful for non-us people who simply don't know our whims. Therefore you're right: "do not violate our guides" is a good approximations for whether we'll even consider it.

However, it might be useful to make note that even if all of the above are matched 100%, we still may decide not to merge it.

**Comment by BlakeWilliams** [comment]
How should we word that? `Even if your TIL matches the above guidelines we may decide not to merge it.`

Not a huge fan of that wording, thoughts?

File: README.md [hunk 3]
```diff
@@ -16,6 +16,23 @@ you might also like
 
 [Upcase]: https://upcase.com?utm_source=til
 
+## Contributing a TIL
```
Comment (JoelQ) [inline README.md]: These should probably be added to `CONTRIBUTING.md` rather than to the README

**Comment by JoelQ** [comment]
> Really it's "we decide whether to merge at our discretion"

I think that's implicit on all open source projects? I think just explicitly stating that violating the guides is an automatic reject is enough. Thoughts @BlakeWilliams ?

File: README.md (hunk 3 above)
Comment (BlakeWilliams) [inline README.md]: I don't think many people will read it if it's in `CONTRIBUTING.md`.

**Comment by BlakeWilliams** [comment]
I think it's fine for now, if we run into any issues we can expand on it.

**Comment by JoelQ** [comment]
I think this is good to merge once it's moved over to `CONTRIBUTING` :ship:

File: README.md (hunk 3 above)
Comment (JoelQ) [inline README.md]: If people can't bother to read `CONTRIBUTING.md` then we can just close their PRs. Github even prompts you to check contribution guidelines when you open a PR 

contributing

Comment (bcjordan) [inline README.md]: Also not unreasonable to point the CONTRIBUTING.md file out as "READ THIS TO CONTRIBUTE" in README.md, (assuming any separation makes sense)

**Comment by ameyjadiye** [comment]
Not sure this is right thread to ask this question but how you guys decide to accept PR, i can see 23 PR's for the repo, many of them are under discussion, i understand ground rules but shall we keep some time constraint to accept/reject PR?

**Comment by BlakeWilliams** [comment]
@JoelQ One more look?

**Comment by JoelQ** [comment]
LGTM :ship:

**Pull request merged**